homeassistant/components/elmo_alarm/__init__.py

```diff
@@ -0,0 +1,237 @@
+                ): vol.All(cv.time_period, cv.positive_timedelta),
+                vol.Required(CONF_STATES): vol.All(cv.ensure_list, [STATE_SCHEMA]),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the e-connect Elmo Alarm platform."""
+
+    conf = config[DOMAIN]
+
+    host = conf[CONF_HOST]
+    vendor = conf[CONF_VENDOR]
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+    states = conf[CONF_STATES]
+    scan_interval = conf[CONF_SCAN_INTERVAL]
+
+    _LOGGER.warning("Scan Interval: %s", scan_interval)
+
+    client = ElmoClientWrapper(host, vendor, username, password, states)
+    await client.update()
+
+    hass.data[DOMAIN] = client
+
+    hass.async_create_task(
+        async_load_platform(
+            hass,
+            "binary_sensor",
+            DOMAIN,
+            {"zones": client.zones, "inputs": client.inputs},
+            config,
+        )
+    )
+
+    hass.async_create_task(
+        async_load_platform(hass, "alarm_control_panel", DOMAIN, {}, config)
+    )
+
+    async def update():
+        _LOGGER.debug("Connecting to e-connect to retrieve states")
+        await client.update()
+        async_dispatcher_send(hass, SIGNAL_ARMING_STATE_CHANGED, client.state)
+
+        for zone in client.zones:
+            async_dispatcher_send(hass, SIGNAL_ZONE_CHANGED, zone)
+
+        for inp in client.inputs:
+            async_dispatcher_send(hass, SIGNAL_INPUT_CHANGED, inp)
+
+        async_call_later(
+            hass,
+            scan_interval.total_seconds(),
+            lambda _: hass.async_create_task(update()),
+        )
+
+    await update()
+
+    return True
+
+
+class ElmoClientWrapper(ElmoClient):
+    """Wrapping the Elmo client class to adapt it to home assistant."""
+
+    def __init__(self, host, vendor, username, password, states_config):
+        """Initialize the elmo client wrapper."""
+
+        self._username = username
+        self._password = password
+        self._states_config = states_config
+        self._data = None
+        self.state = None
+        self.states = None
+        self.zones = None
+        self.inputs = None
+
+        ElmoClient.__init__(self, host, vendor)
+
+    async def update(self):
+        """Get updates and refresh internal states."""
+        try:
+            data = self.check()
+        except PermissionDenied as exception:
+            _LOGGER.warning("Invalid session, trying to authenticate %s", exception)
+            try:
+                self.auth(self._username, self._password)
+                data = self.check()
+            except PermissionDenied as exception:
+                _LOGGER.warning("Invalid credentials: %s", exception)
+            except HTTPError as exception:
+                _LOGGER.warning(
+                    "Got HTTP error when authenticating. Check credentials. Code: %s",
```

Comment:
Remove the credentials part because an HTTPError does not have to caused by incorrect credentials.
```suggestion
                    "Got HTTP error when authenticating: %s",
```

Reply:
Fine. Done.